### DISCUSSION TITLE
Loaded to Reference

### DISCUSSION THREAD

**Discussion by JulianCissen** [mikro-orm/mikro-orm#4811] [Q&A] [answered]
I have a use-case where I need to supply an entity to another entity constructor. The entity needs to be a Loaded entity, so I can do something with the relations in the constructor. I also need to assign the same entity as a reference to a ToOne relation.

What I would think would be the right way to do this, is use the ref wrapper. That's however not possible. The only way I can find out to do this is by working around it using the `createFromPk` static method on the `Reference` class.

Consider the following definitions:
```typescript
import { Entity, Property, PrimaryKey, OneToMany, ManyToOne, Collection, Ref, ref, Loaded, Reference } from "@mikro-orm/core";
import { v4 } from "uuid";

@Entity()
export class Author {
  @PrimaryKey()
  id = v4();

  @Property()
  name: string

  @OneToMany({
    entity: () => Book,
    mappedBy: 'author',
  })
  books = new Collection<Book>(this);

  constructor(name: string) {
    this.name = name;
  }
}

@Entity()
export class Book {
  @PrimaryKey()
  id = v4();

  @Property()
  name: string;

  @ManyToOne({
    entity: () => Author,
    ref: true,
    inversedBy: 'books',
  })
  author: Ref<Author>;

  constructor(name: string; author: Loaded<Author, 'books'>) {
    author.books.getItems().forEach((book) => { console.log(book.name) })
    this.name = name;

    // Won't work (obviously), since author isn't a ref.
    this.author = author;
    // Won't work, since ref wraps the Loaded entity in stead of the plain entity.
    this.author = ref(author);
    // Will work, but seems too verbose to me.
    this.author = Reference.createFromPK(Author, author.id);
  }
}
```

Is this the way this is supposed to be done?

**Comment by B4nan** [comment, marked as answer]
Those things should be working fine in v6, the types are rewritten there. You should be doing `this.author = ref(author)` - just tried and it works fine with v6.

**Reply by JulianCissen** [reply]
Great, thank you!
